pkg/autodiscovery/providers/kube_services.go

```diff
@@ -115,6 +115,11 @@ func (k *KubeServiceConfigProvider) invalidateIfChanged(old, obj interface{}) {
 		k.upToDate = false
 		return
 	}
+	if valuesDiffer(castedObj.Annotations, castedOld.Annotations, kubeEndpointAnnotationPrefix) {
```

Comment:
nitpick: we could probably just compare once on the common `ad.datadoghq.com/` prefix instead of iterating twice

Reply:
comparing `ad.datadoghq.com/` only could lead to create AD-services/endpoints services that are not specified in annotations, it would be better to refactor `valuesDiffer` and makes it supports multiple prefixes checks